### PR TITLE
Documentation for Azure Cache for Redis (Database Access)

### PR DESCRIPTION
**Pull request by greedy52**
Related PRs:
- #16551 
- #16755

### CODE REVIEW

File: docs/pages/includes/database-access/azure-configure-service-principal.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+There are a couple of ways for the Teleport Database service to access Azure
+resources:
```
Comment (greedy52) [inline docs/pages/includes/database-access/azure-configure-service-principal.mdx:2]: moved here without changes except for this introduction line.

**Review by smallinsky — APPROVED** [review]

File: docs/pages/database-access/guides/azure-redis.mdx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+This role definition allows Teleport to discover and list keys for both Azure
+Redis and Azure Redis Enterprise, but Teleport only needs permissions for the
+database types you have. The assignable scopes include a subscription, so the
+role can be assigned at any resource scope within that subscription, or
+assigned using the subscription scope itself.
+
+<Admonition type="note" title="Custom role assignable scope">
+Custom roles, unlike
+[Azure built-in roles](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles),
```
Comment (ptgott) [inline docs/pages/database-access/guides/azure-redis.mdx:129]: I think we should move this link to a "Further reading" section so the user can get to the end of this guide more quickly. Same with the second link in this Admonition. (per the [style guide](https://goteleport.com/docs/contributing/documentation/style-guide/#expectations-1))

File: docs/pages/database-access/guides/azure-redis.mdx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+can not have a root assignable scope. The highest assignable scope that
+can be used in a custom role is subscription scope. Using a management group scope is currently an Azure
+preview feature, and only allows for a single management group in the "assignableScopes" of a role
```
Comment (ptgott) [inline docs/pages/database-access/guides/azure-redis.mdx:132]:
```suggestion
preview feature, and only allows for a single management group in the `assignableScopes` of a role 
```

File: docs/pages/database-access/guides/azure-redis.mdx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+Go to the [Subscriptions](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBlade) page and select a subscription.
+
+Click on *Access control (IAM)* in the subscription and select *Add > Add custom role*:
+<Figure align="left">
+![IAM custom role](../../../img/database-access/guides/azure/add-custom-role@2x.png)
+</Figure>
+
+In the custom role creation page, click the *JSON* tab and click *Edit*, then paste the JSON example
+and replace the subscription in "assignableScopes" with your own subscription id:
```
Comment (ptgott) [inline docs/pages/database-access/guides/azure-redis.mdx:146]:
```suggestion
and replace the subscription in `assignableScopes` with your own subscription id:
```

File: docs/pages/database-access/guides/azure-redis.mdx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+To retrieve credentials for a database and connect to it:
+
+```code
+$ tsh db connect my-azure-redis
+```
+
+Note that the Teleport Database Service will retrieve the access key and
+authenticate with the Redis server on the backend automatically. Therefore,
+`AUTH <access-key>` command is **not required** here once connected.
```
Comment (ptgott) [inline docs/pages/database-access/guides/azure-redis.mdx:210]:
```suggestion
the `AUTH <access-key>` command is **not required** here once connected.
```

**Comment by greedy52** [comment]
Thanks @ptgott ! I've refactored the role definition part a bit. Not sure if this ways is better.

**Review by ptgott — APPROVED** [review]

**Pull request merged**